File: src/main/java/com/euc/grantfitassessment/eval/pipeline/EligibilityCorrectnessFilter.java
package com.euc.grantfitassessment.eval.pipeline;

import com.euc.core.EvaluationFilter;
import com.euc.core.EvaluationStage;
import com.euc.core.PipelineContext;
import com.euc.grantfitassessment.pipeline.GrantFitContextKeys;

/** Scores EUC evaluation stage "eligibilityCorrectness": actual eligibility must match the independently-established ground truth. */
public class EligibilityCorrectnessFilter implements EvaluationFilter {

    @Override
    public Verdict evaluate(PipelineContext context, EvaluationStage stage) {
        boolean actual = context.get(GrantFitContextKeys.ELIGIBLE, Boolean.class);
        boolean expected = context.get(GrantFitEvalContextKeys.EXPECTED_ELIGIBLE, Boolean.class);
        return actual == expected ? Verdict.PASSED : Verdict.FAILED;
    }
}

File: src/euc/grantfitassessment/eval/pipeline/eligibility_correctness_filter.py
"""EVAL-ELIGIBILITY — pure equality against ground truth, no leniency."""

from __future__ import annotations

from euc.core.context import PipelineContext
from euc.core.filters import Verdict
from euc.core.models import EvaluationCriterion
from euc.grantfitassessment import context_keys as prod_keys
from euc.grantfitassessment.eval import context_keys as eval_keys


def eligibility_correctness_filter(context: PipelineContext, criterion: EvaluationCriterion) -> Verdict:
    actual: bool = context.get(prod_keys.ELIGIBLE)
    expected: bool = context.get(eval_keys.EXPECTED_ELIGIBLE)
    return Verdict.PASSED if actual == expected else Verdict.FAILED
